esphome/config_validation.py

```diff
@@ -742,7 +742,9 @@ def validate(obj):
 
         number = sum(k in keys for k in obj)
         if number > 1:
-            raise Invalid(f"Cannot specify more than one of {', '.join(keys)}.")
+            raise Invalid(
+                f"Cannot specify more than one of {', '.join(keys)}.", path=[keys[0]]
+            )
```

Comment:
Badly formatted but something like this
```suggestion
            errs = [Invalid(f"Cannot specify more than one of {', '.join(keys)}.", path=[key]) for key in keys]
            raise MultipleInvalid(errs)
```